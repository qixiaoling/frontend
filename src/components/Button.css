:root {
    --primary: #fff;
}

.btn {
    padding: 8px 20px;
    border-radius: 2px;
    outline: none;
    border: none;
    cursor: pointer;
}

.btn--primary {
    background-color: var(--primary);
    color: #242424;
    border: 1px solid var(--primary);
}
.btn--outline {
    background-color: transparent;
    color: #fff;
    padding: 8px 20px;
    border: 1px solid var(--primary);
    transition: all 0.3s ease-out;
}

.btn--card{
    background-color: #242424;
    color: var(--primary);
    border: 1px solid var(--primary);
    border-radius: 4px;
    transition: ease-out 0.3s;
    border: 3px solid #BEAF6F;
    position: relative;
    z-index:1;
}
.btn--small:hover {
    color: #fff;
    cursor: pointer;
}

.btn--small:before{
    transition: 0.5s all ease;
    position: absolute;
    top:0;
    left:50%;
    right: 50%;
    bottom: 0;
    opacity: 0;
    content: "";
    background-color: #BEAF6F;
}
.btn--small:hover:before {
    transition: 0.5s all ease;
    left:0;
    right:0;
    opacity: 1;
    z-index: -1;
}
.btn--medium {
    padding: 8px 20px;
    font-size: 18px;
}

.btn--large {
    padding: 12px 26px;
    font-size: 20px;
}
.btn--small{
    padding: 8px 20px;
    font-size: 18px;
}


.btn--large:hover,
.btn--medium:hover {
    transition: all 0.3s ease-out;
    background: #fff;
    color: #242424;
    transition: 250ms;
}
.sign-up-button {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}