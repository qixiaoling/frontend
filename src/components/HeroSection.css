/*video {*/
/*    object-fit: cover;*/
/*    width: 100%;*/
/*    height: 100%;*/
/*    position: fixed;*/
/*    z-index: -1;*/
/*}*/

.hero-container {
    background: url('../jasper-geys-NyRe1Mj1pm4-unsplash.jpg') center center/cover no-repeat;
    height: 100vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    box-shadow: inset 0 0 0 1000px rgba(0, 0, 0, 0.2);
    object-fit: contain;
}

h1{
    font-weight: bold;
    font-size: xxx-large;
    color: var(--primary);
}

p{
    font-size: x-large;
    color: var(--primary);
    font-weight: lighter;
}

