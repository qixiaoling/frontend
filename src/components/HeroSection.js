import React from 'react';
import '../App.css';
import { Button } from './Button';
import './HeroSection.css';
import '../road.jpg';
import '../video.mp4';

function HeroSection() {
    return (
        <div className='hero-container'>
            <img src='../road.jpg' />
            <h1>We are profesional</h1>
            <p>We are our customers' go-to!</p>
            <div>
                <Button
                    className='btns'
                    buttonStyle='btn--outline'
                    buttonSize='btn--large'
                >
                    Grow with us
                </Button>
            </div>
        </div>
    );
}

export default HeroSection;