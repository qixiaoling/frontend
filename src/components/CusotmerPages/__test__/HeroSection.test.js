import React from 'react'
import ReactDOM from 'react-dom'
import HeroSection from "../../HeroSection";
import {cleanup} from "@testing-library/react";
import {render} from '@testing-library/react'
import "@testing-library/jest-dom"
import renderer from 'react-test-renderer'
import {Button} from "../../Button";





afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<h1></h1>, div)
})

it("renders title correctly", ()=>{
    const {getByTestId}  = render(<Button></Button>);
    expect(getByTestId('test').value).toEqual("Grow with us")
})

it("match snapshot", ()=>{
    const tree = renderer.create(<Button></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})