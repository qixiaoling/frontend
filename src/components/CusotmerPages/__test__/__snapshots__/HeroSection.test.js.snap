// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<button
  className="btn btn--primary btn--medium"
/>
`;
